kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java

```diff
@@ -128,28 +126,29 @@ public ReadOnlyChunkImpl(
     LOG.info("Created a new read only chunk - zkSlotId: {}", slotId);
   }
 
-  private KaldbMetadataStoreChangeListener cacheNodeListener() {
-    return () -> {
-      CacheSlotMetadata cacheSlotMetadata = cacheSlotMetadataStore.getSync(slotName);
-      Metadata.CacheSlotMetadata.CacheSlotState newSlotState = cacheSlotMetadata.cacheSlotState;
-
-      if (newSlotState.equals(Metadata.CacheSlotMetadata.CacheSlotState.ASSIGNED)) {
-        LOG.info("Chunk - ASSIGNED received");
-        if (!cacheSlotLastKnownState.equals(Metadata.CacheSlotMetadata.CacheSlotState.FREE)) {
-          LOG.warn(
-              "Unexpected state transition from {} to {}", cacheSlotLastKnownState, newSlotState);
-        }
-        executorService.execute(() -> handleChunkAssignment(cacheSlotMetadata));
-      } else if (newSlotState.equals(Metadata.CacheSlotMetadata.CacheSlotState.EVICT)) {
-        LOG.info("Chunk - EVICT received");
-        if (!cacheSlotLastKnownState.equals(Metadata.CacheSlotMetadata.CacheSlotState.LIVE)) {
-          LOG.warn(
-              "Unexpected state transition from {} to {}", cacheSlotLastKnownState, newSlotState);
+  private KaldbMetadataStoreChangeListener<CacheSlotMetadata> cacheNodeListener() {
+    return (metadata) -> {
```

Comment:
nice! So we don't even need to call `getSync(slotId/slotName)` in this impl. 

We listen to all events changed and if the event matches the one we care about we execute some code.